### ISSUE TITLE
Scripting when user has joined team, ie listening for team_join events

### ISSUE THREAD

**Issue by jschuur** [slackhq/hubot-slack#196]
Should I be able to use [this](https://github.com/smallmovesclub/smallbot/blob/master/scripts/event.coffee) to listen to a [team_join](https://api.slack.com/events/team_join) event?

```
module.exports = (robot) ->
	robot.on  "team_join", (user) ->
		console.log "Triggered team_join event"
		robot.logger.debug "Triggered team_join event"
```

This doesn't seem to be getting triggered and there's [some discussion](https://github.com/slackhq/node-slack-client/issues/40#issuecomment-103074322) by @khanh-nguyen about API changes related to using `user_change`, but that's not working for me either.

In a nutshell, I'm looking for a way for a Slack adapter hubot to trigger logic when a user accepts their Slack invite.

**Comment by technicalpickles** [comment]
The adapter has an instance of a [node-slack client](https://github.com/slackhq/node-slack) as `client`, which does get events for that. You can access it at `robot.adapter.client`, so you should be able to add a your own handler for team_join.

I'd have to think about it more, but this might be able to be generalized for hubot core.

**Comment by ShimiTaNaka** [comment]
@jschuur If you want, you can use: 
`robot.adapter.client.on "raw_message", (message) ->
    switch (message.type)
      when "team_join"
        console.log "Triggered team_join event"
`
The node client emits a message_raw event regardless of the event itself.

**Comment by jschuur** [comment]
Thanks, @ShimiTaNaka! Haven't tried this yet (I ended up building something using [slack-bot-api](https://github.com/mishk0/slack-bot-api)), but it looks like what I was originally asking for.
